pkg/proxy/nftables/proxier.go

```diff
@@ -308,25 +308,83 @@ func NewDualStackProxier(
 	return metaproxier.NewMetaProxier(ipv4Proxier, ipv6Proxier), nil
 }
 
-// nftablesBaseChains lists our "base chains"; those that are directly connected to the
-// netfilter hooks (e.g., "postrouting", "input", etc.), as opposed to "regular" chains,
-// which are only run when a rule jumps to them. See
-// https://wiki.nftables.org/wiki-nftables/index.php/Configuring_chains.
+// Packet flow through netfilter looks something like:
 //
-// These are set up from setupNFTables() and then not directly referenced by
-// syncProxyRules().
+//                            +==================+
+//                            |   local process  |
+//                            +==================+
+//                              ^              |*
+//                              |              v
+//                          +-------+      +--------+
+//                          | input |      | output |
+//                          +-------+      +--------+
+//                              ^              |
+// +---------+   +------------+ |  +---------+ v  +-------------+   +--------+
+// | ingress |-->| prerouting |-*->| forward |-*->| postrouting |-->| egress |
+// +---------+   +------------+    +---------+    +-------------+   +--------+
 //
-// All of our base chains have names that are just "${type}-${hook}". e.g., "nat-prerouting".
+// (where the "*"s represent routing decisions, and all of the boxes except "local
+// process" represent netfilter hooks. More detailed versions of this diagram can be seen
+// at https://en.wikipedia.org/wiki/Netfilter#/media/File:Netfilter-packet-flow.svg and
+// https://wiki.nftables.org/wiki-nftables/index.php/Netfilter_hooks .) Note that despite
+// how it looks, packets cannot actually pass through the "local process" graph node;
+// Packets on the "input path" terminate at "local process", and packets on the "output
+// path" originate there (while packets on the "forward path" bypass it).
+//
+// From netfilter's perspective, there is no real difference between a NIC connected to an
+// ethernet network, and a veth connected to another network namespace on the same host.
+// In both cases, packets arriving via that interface come in via ingress and prerouting,
+// and packets that get routed to that interface leave via postrouting and egress. Thus,
+// "local process" in the diagram more specifically means "host-network process".
+//
+// Kube-proxy uses the netfilter hooks as follows:
+//
+//   - We do the DNAT for inbound traffic in "prerouting": this covers traffic coming from
+//     off-node to service IPs, and traffic coming from pods to service IPs. (We *must* do
```

Comment:
being pedantic, do you want to differentiate between clusterip, nodeport, loadbalancerip and externalip?